db/schema_tables.cc

```diff
@@ -1034,8 +1031,8 @@ future<std::vector<user_type>> create_types(replica::database& db, const std::ve
             return r->get_nonnull<sstring>("keyspace_name") != keyspace;
         });
         auto ks = db.find_keyspace(keyspace).metadata();
-        auto v = co_await create_types(*ks, boost::make_iterator_range(i, next) | boost::adaptors::indirected);
-        ret.insert(ret.end(), std::make_move_iterator(v.begin()), std::make_move_iterator(v.end()));
+        auto v = co_await create_types(*ks, std::ranges::subrange(i, next) | std::views::transform([] (auto&& r) { return *r; }));
```

Comment:
Not too important, but I think the transformation isn't exact. I expect boost::adaptors::indirected returns a reference, while your lambda returns a prvalue or however (I wrote prvalue randomly, I have no idea what it really means).

Reply:
i think it's indeed prvalue. see https://en.cppreference.com/w/cpp/language/value_category#prvalue . and it's equivalent to use `auto` in function declaration, see https://en.cppreference.com/w/cpp/language/auto#Function_declarations we are using rule of template argument deduction, see https://eel.is/c++draft/temp.deduct.call#3 . 

> If P is a cv-qualified type, the top-level cv-qualifiers of P's type are ignored for type deduction[.](https://eel.is/c++draft/temp.deduct.call#3.sentence-1)
> If P is a reference type, the type referred to by P is used for type deduction[.](https://eel.is/c++draft/temp.deduct.call#3.sentence-2)

so the return value's type of the lambda was `query::result_set_row` instead of `const query::result_set_row&`.